sklearn/datasets/tests/test_openml.py

```diff
@@ -0,0 +1,256 @@
+                              expected_observations, expected_features,
+                              expect_sparse=False)
+
+
+def test_fetch_openml_anneal(monkeypatch):
+    # classification dataset with numeric and categorical columns
+    data_id = 2
+    data_name = 'anneal'
+    data_version = 1
+    # Not all original instances included for space reasons
+    expected_observations = 11
+    expected_features = 38
+    _monkey_patch_webbased_functions(monkeypatch, data_id)
+    fetch_dataset_from_openml(data_id, data_name, data_version,
+                              expected_observations, expected_features,
+                              expect_sparse=False)
+
+
+def test_fetch_openml_cpu(monkeypatch):
+    # regression dataset with numeric and categorical columns
+    data_id = 561
+    data_name = 'cpu'
+    data_version = 1
+    expected_observations = 209
+    expected_features = 7
+    _monkey_patch_webbased_functions(monkeypatch, data_id)
+    fetch_dataset_from_openml(data_id, data_name, data_version,
+                              expected_observations, expected_features,
+                              expect_sparse=False)
+
+
+def test_fetch_openml_australian(monkeypatch):
+    # sparse dataset
+    # Australian is the only sparse dataset that is reasonably small
+    # as it is inactive, we need to catch the warning. Due to mocking
+    # framework, it is not deactivated in our tests
+    data_id = 292
+    data_name = 'Australian'
+    data_version = 1
+    # Not all original instances included for space reasons
+    expected_observations = 85
+    expected_features = 14
+    _monkey_patch_webbased_functions(monkeypatch, data_id)
+    assert_warns_message(
+        UserWarning,
+        "Version 1 of dataset Australian is inactive,",
+        fetch_dataset_from_openml,
+        **{'data_id': data_id, 'data_name': data_name,
+           'data_version': data_version,
+           'expected_observations': expected_observations,
+           'expected_features': expected_features,
+           'expect_sparse': False}
+    )
+
+
+def test_fetch_openml_miceprotein(monkeypatch):
+    data_id = 40966
+    data_name = 'MiceProtein'
+    data_version = 4
+    # Not all original instances included for space reasons
+    expected_observations = 7
+    expected_features = 77
+    _monkey_patch_webbased_functions(monkeypatch, data_id)
+    data = fetch_dataset_from_openml(data_id, data_name, data_version,
+                                     expected_observations, expected_features,
+                                     expect_sparse=False)
+    # JvR: very important check, as this dataset defined several row ids
+    # and ignore attributes. Note that data_features json has 82 attributes,
+    # and row id (1), ignore attributes (3) have been removed (and target is
+    # stored in data.target)
+    assert (data.data.shape == (expected_observations, expected_features))
+    assert (data.target.shape == (expected_observations, ))
```

Comment:
Isn't this already tested in the `fetch_dataset_from_openml` function?

Reply:
True, I removed it